org.eclipse.jdt.core.tests.compiler/src/org/eclipse/jdt/core/tests/compiler/parser/ComplianceDiagnoseTest.java

```diff
@@ -1967,112 +1969,48 @@ public void test0042() {
 		"}\n"
 	};
 
-	String expected13ProblemLog =
-			"----------\n" +
-			"1. ERROR in X.java (at line 1)\n" +
-			"	void ___eval() {\n" +
-			"	^^^^^^^^^^^^^^\n" +
-			"Syntax error on tokens, delete these tokens\n" +
-			"----------\n" +
-			"2. ERROR in X.java (at line 2)\n" +
-			"	new Runnable() {\n" +
-			"		int ___run() throws Throwable {\n" +
-			"			return blah;\n" +
-			"		}\n" +
-			"		private String blarg;\n" +
-			"		public void run() {\n" +
-			"		}\n" +
-			"	};\n" +
-			"}\n" +
-			"	^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n" +
-			"Syntax error on tokens, delete these tokens\n" +
-			"----------\n";
-	String expected14ProblemLog =
-		expected13ProblemLog;
-
-	String expected15ProblemLog =
-			"----------\n" +
-			"1. ERROR in X.java (at line 1)\n" +
-			"	void ___eval() {\n" +
-			"	^^^^\n" +
-			"Syntax error on token \"void\", @ expected\n" +
-			"----------\n" +
-			"2. ERROR in X.java (at line 1)\n" +
-			"	void ___eval() {\n" +
-			"	             ^\n" +
-			"Syntax error on token \")\", delete this token\n" +
-			"----------\n" +
-			"3. ERROR in X.java (at line 9)\n" +
-			"	};\n" +
-			"}\n" +
-			"	^^^^\n" +
-			"Syntax error on tokens, delete these tokens\n" +
-			"----------\n" +
-			"4. ERROR in X.java (at line 23)\n" +
-			"	}\n" +
-			"	^\n" +
-			"Syntax error, insert \"}\" to complete ClassBody\n" +
-			"----------\n" +
-			"5. ERROR in X.java (at line 23)\n" +
-			"	}\n" +
-			"	^\n" +
-			"Syntax error, insert \"}\" to complete MemberValue\n" +
-			"----------\n" +
-			"6. ERROR in X.java (at line 23)\n" +
-			"	}\n" +
-			"	^\n" +
-			"Syntax error, insert \")\" to complete Modifiers\n" +
-			"----------\n" +
-			"7. ERROR in X.java (at line 23)\n" +
-			"	}\n" +
-			"	^\n" +
-			"Syntax error, insert \"enum Identifier\" to complete EnumHeader\n" +
-			"----------\n" +
-			"8. ERROR in X.java (at line 23)\n" +
-			"	}\n" +
-			"	^\n" +
-			"Syntax error, insert \"EnumBody\" to complete CompilationUnit\n" +
-			"----------\n";
-
-	String expectedJ14ProblemLog =
-			"----------\n" +
-			"1. ERROR in X.java (at line 1)\n" +
-			"	void ___eval() {\n" +
-			"	^^^^\n" +
-			"Syntax error on token \"void\", record expected\n" +
-			"----------\n" +
-			"2. ERROR in X.java (at line 2)\n" +
-			"	new Runnable() {\n" +
-			"	^^^\n" +
-			"Syntax error on token \"new\", record expected\n" +
-			"----------\n";
+	String expected13ProblemLog = """
+			----------
+			1. ERROR in X.java (at line 1)
+				void ___eval() {
+				^
+			The preview feature Unnamed Classes and Instance Main Methods is only available with source level 21 and above
```

Comment:
It may not make sense to throw this error when compiling at a level below 21. Please handle this to throw an appropriate error

Reply:
It can make sense for the scenario of a user who's using Unnamed Classes and has not configured the project to Java 21 + early-access features. This problem is a way to remind them that maybe what they want is newer Java.

Reply:
@mickaelistria let's say I am compiling at 17 and if I get this error then does it make sense to throw an error with a java language feature much later? Please change

Reply:
Only one error is reported if you try to create an unnamed class when the project is set to Java 17 (or any Java less than 21). A quick fix is presented which can update the Java version of the project and enable preview features.